### PR TITLE
[feat] Add wrapper support for pNFT transfers

### PR DESCRIPTION
**Pull request by ngundotra**
Makes `mpl-token-metadata` compatible with additional accounts request Transfer spec.

Edit: 
- ~missing test for when `pNFT` auth rules is non-null 🤔~ Fixed by cloning in `eBJLFYPxJmMGKuFwpDWkzxZeUrad92kZRC5BJLpzyT9`
- need to finish pass-through tfer for `token-metadata`

### CODE REVIEW

File: tests/itoken-poc.ts [hunk 1]
```diff
@@ -24,27 +19,43 @@ import {
...
+    { commitment: "confirmed" }
+  );
 
-  let returnDataTuple = simulationResult.value.returnData;
-  let [b64Data, encoding] = returnDataTuple["data"];
-  if (encoding !== "base64") {
-    throw new Error("Unsupported encoding: " + encoding);
-  }
+  // When the simulation RPC response is fixed, then the following code will work
+  // but until then, we have to parse the logs manually
```
Comment (joncinque) [inline tests/itoken-poc.ts:49]: Could just fill to 1024 bytes!

File: programs/token-wrapper/src/lib.rs [hunk 2]
```diff
@@ -135,6 +314,7 @@ pub mod token_wrapper {
 enum CalleeProgram {
     TokenStar,
```
Comment (joncinque) [inline programs/token-wrapper/src/lib.rs:315]: nit: Could you rename this to `SplToken`? I was a bit confused, but that might just be because we called `Token*` the interface version for a time

File: programs/token-wrapper/src/lib.rs [hunk 3]
```diff
@@ -135,6 +314,7 @@ pub mod token_wrapper {
...
     Interface,
```
Comment (joncinque) [inline programs/token-wrapper/src/lib.rs:316]: nit: maybe `TransferInterface`?

File: programs/token-wrapper/src/lib.rs [hunk 4]
```diff
@@ -74,6 +87,116 @@ pub mod token_wrapper {
                 set_return_data(&return_data);
                 Ok(())
             }
+            CalleeProgram::TokenMetadata => {
+                let meta = Metadata::from_account_info(&mint.to_account_info())?;
```
Comment (joncinque) [inline programs/token-wrapper/src/lib.rs:91]: It's very clever to make the metadata into the "mint" for the transfer, since you're sure that it has the data. 

How is that meant to be communicated to users of the wrapper program though? Is it one of those "you just have to know" things? Is it possible to figure out that a mint is managed by the metadata program based on some qualities of the token mint?

**Review by joncinque — COMMENTED** [review]
Just one question about how this is meant to be used, otherwise the concept looks sound

File: programs/token-wrapper/src/lib.rs (hunk 3 above)
Comment (ngundotra) [reply programs/token-wrapper/src/lib.rs:316]: yup that's a better name 😂

File: programs/token-wrapper/src/lib.rs (hunk 4 above)
Comment (ngundotra) [reply programs/token-wrapper/src/lib.rs:91]: I don't think it's possible to figure out `mint` is actually owned by `token-metadata`, since you need to fetch ownership account's data to check that.

Comment (joncinque) [reply programs/token-wrapper/src/lib.rs:91]: Oh wait hold on! I found https://github.com/metaplex-foundation/metaplex-program-library/blob/master/token-metadata/program/ProgrammableNFTGuide.md, and if I'm reading it right, if the mint and freeze authority are a particular PDA, then you know it's managed by Metaplex!

Comment (joncinque) [reply programs/token-wrapper/src/lib.rs:91]: And what's pretty cool, this is an example that would require *two* calls to figure out the required accounts, since the first call would just give you *more* accounts, and the second would allow you to derive everything

Comment (ngundotra) [reply programs/token-wrapper/src/lib.rs:89]: Honestly it might be better to just rename the `mint` account to `descriptor` or something more generic
